File: 04_hmac/base64/main.java
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import java.nio.charset.StandardCharsets;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

class Main {
    public static void main(String[] args) throws NoSuchAlgorithmException, InvalidKeyException {
        String text = "Hello world, mac was here";
        String secret = "Secret used to sign hash";

        Mac mac = Mac.getInstance("HmacSHA256");
        mac.init(new SecretKeySpec(secret.getBytes(StandardCharsets.UTF_8), "HmacSHA256"));
        String hash = Base64.getEncoder().encodeToString(mac.doFinal(text.getBytes(StandardCharsets.UTF_8)));

        System.out.println(hash);
    }
}

File: 04_hmac/base64/main.cs
var text = "Hello world, mac was here";
var secret = "Secret used to sign hash";

var hash = Convert.ToBase64String(new System.Security.Cryptography.HMACSHA256(System.Text.Encoding.UTF8.GetBytes(secret)).ComputeHash(System.Text.Encoding.UTF8.GetBytes(text)));

Console.WriteLine(hash);
